spec/rubocop/cop/style/accessor_grouping_spec.rb

```diff
@@ -141,6 +141,47 @@ class Foo
       RUBY
     end
 
+    it 'does not register an offense for grouped accessors below a typechecked accessor method' do
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          extend T::Sig
+
+          sig { returns(Integer) }
+          attr_reader :one
+
+          attr_reader :two, :three
+        end
+      RUBY
+    end
+
+    it 'registers an offense for grouped accessors distinct from a typechecked accessor method' do
+      expect_offense(<<~RUBY)
+        class Foo
+          extend T::Sig
+
+          sig { returns(Integer) }
+          attr_reader :one
+
+          attr_reader :two, :three
+          ^^^^^^^^^^^^^^^^^^^^^^^^ Group together all `attr_reader` attributes.
+
+          attr_reader :four
+          ^^^^^^^^^^^^^^^^^ Group together all `attr_reader` attributes.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
```

Comment:
I ran this test on RuboCop `master` (v1.48.0) and the autocorrect is completely bugged because of the sibling grouping detection. Sorry for not thinking about this sooner! 😳

```
  1) RuboCop::Cop::Style::AccessorGrouping when EnforcedStyle is grouped registers an offense for grouped accessors distinct from a typechecked accessor method
     Failure/Error: expect(new_source).to eq(correction)
     
       expected: "class Foo\n  extend T::Sig\n\n  sig { returns(Integer) }\n  attr_reader :one\n\n  attr_reader :two, :three, :four\nend\n"
            got: "class Foo\n  extend T::Sig\n\n  sig { returns(Integer) }\n  attr_reader :one\nend\n"
     
       (compared using ==)
     
       Diff:
       @@ -3,7 +3,5 @@
        
          sig { returns(Integer) }
          attr_reader :one
       -
       -  attr_reader :two, :three, :four
        end
```